src/transformers/utils/quantizers.py

```diff
@@ -0,0 +1,869 @@
+            )
+            self.quantization_method = QuantizationMethod.BITS_AND_BYTES
+        else:
+            self.quantization_config = None  # no quantization from args
+
+        if self.quantization_method == QuantizationMethod.BITS_AND_BYTES:
+            bnb_quantization_config_kwargs = {
+                k: v for k, v in kwargs.items() if k in inspect.signature(BitsAndBytesConfig).parameters
+            }
+            if bnb_quantization_config_kwargs:
+                raise ValueError(
+                    "You can't pass `load_in_8bit` or any other `BitsAndBytesConfig` argument as a kwarg when passing "
+                    "`quantization_config` argument at the same time."
+                )
+        # if self.quantization_method == QuantizationMethod.AWQ:
+        #     raise ValueError(
+        #         "You cannot quantize with AWQ a non-quantized model using transformers, please refer to the quantization documentation"
+        #         " to read more about how to quantize models with AWQ algorithm https://huggingface.co/docs/transformers/main_classes/quantization"
+        #     )
+
+        return kwargs
+
+    def validate_quantization_config(self, quantization_config):
+        """
+        checks quantization_config class
+        converts dict format into QuantizationConfigMixin
+        returns quantization_config as QuantizationConfigMixin
+        """
+
+        if isinstance(quantization_config, QuantizationConfigMixin):
+            return quantization_config
+
+        elif isinstance(quantization_config, dict):
+            quant_method = quantization_config.get("quant_method", None)
+
+            if quant_method is None:
+                if quantization_config.get("load_in_8bit", False) or quantization_config.get("load_in_4bit", False):
+                    logger.warning(
+                        "The model's quantization_config has no `quant_method` attribute, but has `load_in_8bit` or `load_in_4bit` parameter set. "
+                        "It is recommended to pass quantization config as `BitsAndBytesConfig`. "
+                    )
+                    quant_method = QuantizationMethod.BITS_AND_BYTES
+                else:
+                    raise ValueError(
+                        "The model's quantization config from the arguments has no `quant_method` attribute."
+                    )
+
+            if quant_method == QuantizationMethod.BITS_AND_BYTES:
+                config_class = BitsAndBytesConfig
+            elif quant_method == QuantizationMethod.GPTQ:
+                config_class = GPTQConfig
+            elif quant_method == QuantizationMethod.AWQ:
+                config_class = AwqConfig
+            else:
+                # unknown quantization_method, which is not None:
+                raise NotImplementedError(
+                    f"Unsupported quantization method detected: {quant_method}. Check for updates."
+                )
```

Comment:
what's strange to me is that you have an exact mapping from config to method, the method is just a string basically. I think we should follow what we have in the AutoConfig for example, where you just have `model_type` (here will be the `quant_type` or whatever) and just have a mapping from the string type to the class. 

There's a lot of complexity IMO for something pretty simple + we already commonly use AutoConfigs quite often. The `QuantizationConfig` should inherit form `PushToHubMixin` and have the necessary logic. (be a dataclass we should not need to parse). 
Let's try to re-use what we are familiar with! `GenerationConfig` should help.

Reply:
Then each config will have a type, mapping type to the quantizer kind of like models.

Reply:
I agree that it sould be simplified, just have not figured the best way yet. Will try with your hints.

Not sure about inheriting from `PushToHubMixin` - the HFQuantizer will not wrap the model (as Peft does, but will be used as an external tool to process the model).

Reply:
refactored this part. Now checking `quantization_config` type instead.